### PR TITLE
minor: Use readable names for TaskNode display

### PR DESCRIPTION
**Pull request by fediazgon**
Signed-off-by: Fernando Diaz <fediazgon@gmail.com>

# TL;DR
As described in https://github.com/flyteorg/flyte/issues/1242. UI should show readable names to the user instead of execution ids.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

Currently, Flyte Console shows [NodeExecutionIdentifier#node_id](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/identifier.proto#L53) which can be auto-generated and not readable for the user. Instead, I propose to use (Compiled)[Node#id](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/workflow.proto#L103) which is usually set to a readable name and can be overriden the user (using `with_overrides(node_name=....`) in `flytekit-python` and `builder.apply(/*nodeId=*/...,...)` in `flytekit-java`.

I've also improved the lookup nodes code to also lookup for nodes inside branches.

This is the before and after for a complex workflow:

Before:

Screenshot 2021-11-23 at 10 31 15

After

Screenshot 2021-11-23 at 11 23 44

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/1242

### CODE REVIEW

**Comment by fediazgon** [comment]
On a side not. I had to add the following change (which is not staged) to make my local deployment work.

Screenshot 2021-11-23 at 11 05 59
 
 Otherwise, I would get the error:
 
 ```
 TypeError: Cannot read properties of undefined (reading 'error')
 ```

**Comment by fediazgon** [comment]
Ping @jsonporter

**Review by jsonporter — DISMISSED** [review]
There is a failing test but otherwise this lgtm; fix the test and i'll approve 👍

**Comment by fediazgon** [comment]
Sorry, @jsonporter. Now, `make test_unit` ran successfully.

**Review by jsonporter — APPROVED** [review]
lgmt
